File: multiple-languages/go/alicloudroscdkarms/IEnvironmentFeature.go
package alicloudroscdkarms

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkarms/internal"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
)

// Represents a `EnvironmentFeature`.
type IEnvironmentFeature interface {
	alicloudroscdkcore.IResource
	// Attribute EnvironmentId: The environment ID.
	AttrEnvironmentId() interface{}
	// Attribute Feature: The installation information of the feature.
	AttrFeature() interface{}
	// Attribute FeatureName: The name of the feature.
	AttrFeatureName() interface{}
	// Attribute FeatureStatus: The status of the feature.
	AttrFeatureStatus() interface{}
	Props() *EnvironmentFeatureProps
}

// The jsii proxy for IEnvironmentFeature
type jsiiProxy_IEnvironmentFeature struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IEnvironmentFeature) AttrEnvironmentId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrEnvironmentId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvironmentFeature) AttrFeature() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrFeature",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvironmentFeature) AttrFeatureName() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrFeatureName",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvironmentFeature) AttrFeatureStatus() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrFeatureStatus",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvironmentFeature) Props() *EnvironmentFeatureProps {
	var returns *EnvironmentFeatureProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Arms/AlibabaCloud/SDK/ROS/CDK/Arms/IEnvironmentFeature.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Arms
{
    /// <summary>Represents a `EnvironmentFeature`.</summary>
    [JsiiInterface(nativeType: typeof(IEnvironmentFeature), fullyQualifiedName: "@alicloud/ros-cdk-arms.IEnvironmentFeature")]
    public interface IEnvironmentFeature : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute EnvironmentId: The environment ID.</summary>
        [JsiiProperty(name: "attrEnvironmentId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrEnvironmentId
        {
            get;
        }

        /// <summary>Attribute Feature: The installation information of the feature.</summary>
        [JsiiProperty(name: "attrFeature", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrFeature
        {
            get;
        }

        /// <summary>Attribute FeatureName: The name of the feature.</summary>
        [JsiiProperty(name: "attrFeatureName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrFeatureName
        {
            get;
        }

        /// <summary>Attribute FeatureStatus: The status of the feature.</summary>
        [JsiiProperty(name: "attrFeatureStatus", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrFeatureStatus
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-arms.EnvironmentFeatureProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Arms.IEnvironmentFeatureProps Props
        {
            get;
        }

        /// <summary>Represents a `EnvironmentFeature`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IEnvironmentFeature), fullyQualifiedName: "@alicloud/ros-cdk-arms.IEnvironmentFeature")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Arms.IEnvironmentFeature
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute EnvironmentId: The environment ID.</summary>
            [JsiiProperty(name: "attrEnvironmentId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrEnvironmentId
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute Feature: The installation information of the feature.</summary>
            [JsiiProperty(name: "attrFeature", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrFeature
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute FeatureName: The name of the feature.</summary>
            [JsiiProperty(name: "attrFeatureName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrFeatureName
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute FeatureStatus: The status of the feature.</summary>
            [JsiiProperty(name: "attrFeatureStatus", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrFeatureStatus
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-arms.EnvironmentFeatureProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Arms.IEnvironmentFeatureProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Arms.IEnvironmentFeatureProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}
